packages/flutter/lib/src/widgets/binding.dart

```diff
@@ -492,6 +492,25 @@ mixin WidgetsBinding
       _debugShowWidgetInspectorOverrideNotifierObject ??= ValueNotifier<bool>(false);
   ValueNotifier<bool>? _debugShowWidgetInspectorOverrideNotifierObject;
 
+  /// If true, [WidgetInspector] will not be automatically injected into the
+  /// widget tree.
+  ///
+  /// This overrides the behavior where [WidgetInspector] is added to the
+  /// widget tree created by [WidgetsApp] when the `debugShowWidgetInspector`
+  /// value is set in [WidgetsApp] or [debugShowWidgetInspectorOverride] is set
+  /// to true.
+  ///
+  /// Used by tools that want more control over which widgets can be selected
+  /// and highlighted by the widget inspector by manually injecting instance of
+  /// [WidgetInspector].
+  bool get debugWillManuallyInjectWidgetInspector => _debugWillManuallyInjectWidgetInspector;
```

Comment:
I think a better name might be `debugExcludeWidgetInspector` or something like that. All this property does is prevent Flutter from injecting the widget inspector, I don't think we necessarily need to have the naming indicate a promise that some other tool will inject it.

Reply:
That's a good point. Maybe `debugExcludeRootWidgetInspector` would be even more clear?

Reply:
Sure, that SGTM!